### PR TITLE
Is old storage endpoint development

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
proper implementation of #328

### CODE REVIEW

File: process/nodeGroupProcessor_test.go [hunk 1]
```diff
@@ -512,3 +515,119 @@ func TestHeartbeatProcessor_NoDataForAShardShouldNotUpdateCache(t *testing.T) {
...
+			},
+			&mock.HeartbeatCacherMock{},
+			10,
+		)
+
+		_, err := proc.IsOldStorageForToken("token", 37)
+		require.Equal(t, process.ErrSendingRequest, err)
+	})
+
+	t.Run("some observers fail, should return error", func(t *testing.T) {
```
Comment (andreibancioiu) [inline process/nodeGroupProcessor_test.go:545]: :+1:

File: process/nodeGroupProcessor_test.go [hunk 2]
```diff
@@ -512,3 +515,119 @@ func TestHeartbeatProcessor_NoDataForAShardShouldNotUpdateCache(t *testing.T) {
...
+			10,
+		)
+
+		isOldStorage, err := proc.IsOldStorageForToken("token", 37)
+		require.True(t, isOldStorage)
+		require.NoError(t, err)
+	})
+}
+
+func TestComputeTokenStorageKey(t *testing.T) {
```
Comment (andreibancioiu) [inline process/nodeGroupProcessor_test.go:624]: :+1:

**Review by andreibancioiu — APPROVED** [review]

File: api/groups/baseNodeGroup.go [hunk 3]
```diff
@@ -27,6 +28,7 @@ func NewNodeGroup(facadeHandler data.FacadeHandler) (*nodeGroup, error) {
 
 	baseRoutesHandlers := []*data.EndpointHandlerData{
 		{Path: "/heartbeatstatus", Handler: ng.getHeartbeatData, Method: http.MethodGet},
+		{Path: "/old-storage-token/:token/nonce/:nonce", Handler: ng.isOldStorageForToken, Method: http.MethodGet},
```
Comment (iulianpascalau) [inline api/groups/baseNodeGroup.go:31]: this is not related to node group

File: facade/baseFacade.go [hunk 4]
```diff
@@ -425,6 +425,11 @@ func (epf *ElrondProxyFacade) GetLastPoolNonceForSender(sender string) (uint64,
 	return epf.txProc.GetLastPoolNonceForSender(sender)
 }
 
+// IsOldStorageForToken returns true is the storage for a given token is old
+func (epf *ElrondProxyFacade) IsOldStorageForToken(tokenID string, nonce uint64) (bool, error) {
+	return epf.heartbeatProc.IsOldStorageForToken(tokenID, nonce)
```
Comment (iulianpascalau) [inline facade/baseFacade.go:430]: this is totally unreleated

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: api/groups/baseNodeGroup.go (hunk 3 above)
Comment (bogdan-rosianu) [reply api/groups/baseNodeGroup.go:31]: added a TODO to remove it

File: facade/baseFacade.go (hunk 4 above)
Comment (bogdan-rosianu) [reply facade/baseFacade.go:430]: added a TODO to remove it

**Review by iulianpascalau — APPROVED** [review]

**Review by andreibancioiu — DISMISSED** [review]

File: cmd/proxy/main.go [hunk 5]
```diff
@@ -485,7 +485,7 @@ func createVersionsRegistry(
 	htbCacher := cache.NewHeartbeatMemoryCacher()
 	cacheValidity := time.Duration(cfg.GeneralSettings.HeartbeatCacheValidityDurationSec) * time.Second
 
-	htbProc, err := process.NewHeartbeatProcessor(bp, htbCacher, cacheValidity)
+	htbProc, err := process.NewNodeGroupProcessor(bp, htbCacher, cacheValidity)
```
Comment (sstanculeanu) [inline cmd/proxy/main.go:488]:
```suggestion
	nodeProc, err := process.NewNodeGroupProcessor(bp, htbCacher, cacheValidity)
```

File: facade/baseFacade.go [hunk 6]
```diff
@@ -32,7 +32,7 @@ type ElrondProxyFacade struct {
 	accountProc      AccountProcessor
 	txProc           TransactionProcessor
 	scQueryService   SCQueryService
-	heartbeatProc    HeartbeatProcessor
+	heartbeatProc    NodeGroupProcessor
```
Comment (sstanculeanu) [inline facade/baseFacade.go:35]:
```suggestion
	nodeProc    NodeGroupProcessor
```

File: facade/baseFacade.go [hunk 7]
```diff
@@ -51,7 +51,7 @@ func NewElrondProxyFacade(
 	accountProc AccountProcessor,
 	txProc TransactionProcessor,
 	scQueryService SCQueryService,
-	heartbeatProc HeartbeatProcessor,
+	heartbeatProc NodeGroupProcessor,
```
Comment (sstanculeanu) [inline facade/baseFacade.go:54]:
```suggestion
	nodeProc NodeGroupProcessor,
```

File: facade/mock/heartbeatProcessorStub.go [hunk 8]
```diff
@@ -4,7 +4,13 @@ import "github.com/ElrondNetwork/elrond-proxy-go/data"
 
 // HeartbeatProcessorStub represents a stub implementation of a HeartbeatProcessor
 type HeartbeatProcessorStub struct {
-	GetHeartbeatDataCalled func() (*data.HeartbeatResponse, error)
+	GetHeartbeatDataCalled     func() (*data.HeartbeatResponse, error)
```
Comment (sstanculeanu) [inline facade/mock/heartbeatProcessorStub.go:7]: stub should be renamed as well to NodeProcessorStub

File: versions/factory/versionedFacadeCreator.go [hunk 9]
```diff
@@ -19,7 +19,7 @@ type FacadeArgs struct {
 	FaucetProcessor              facade.FaucetProcessor
 	BlockProcessor               facade.BlockProcessor
 	BlocksProcessor              facade.BlocksProcessor
-	HeartbeatProcessor           facade.HeartbeatProcessor
+	HeartbeatProcessor           facade.NodeGroupProcessor
```
Comment (sstanculeanu) [inline versions/factory/versionedFacadeCreator.go:22]:
```suggestion
	NodeProcessor           facade.NodeGroupProcessor
```

**Review by andreibancioiu — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]
